policy-controller/src/index_list.rs

```diff
@@ -53,9 +46,14 @@ impl<A, T> IndexList<A, T> {
     }
 }
 
-// This constructor is more ergonomic to use than IndexList::<A, A>::new because
-// the compiler is more easily able to infer A and doesn't require callers to
-// fill in explicit type parameters.
-pub fn new<A>(index: Arc<RwLock<A>>) -> IndexList<A, A> {
-    IndexList::<A, A>::new(index)
+impl<A> IndexList<A> {
+    /// Returns a new `IndexList`.
+    ///
+    /// The second type parameter in the return value here can be anything that
+    /// implements `IndexNamespacedResource<R>`, since it will just be `None`.
+    /// Ideally, the type should be `!` (bottom) but `A` is conveniently available,
+    /// so we use that.
```

Comment:
now that there's a default type parameter, is this comment still relevant?

Reply:
I was a bit confused by the comment since I didn't look in detail at the previous impl (before some of your comments got resolved 😅 ). Also, what do we mean by

> type should be `!` (bottom).

Reply:
ah, okay, it's because we returned an `IndexList<A, A>`.

Reply:
I think so, it's talking about the somewhat surprising type parameters in the return type.